core/contracts/test/MockAdmin.sol

```diff
@@ -0,0 +1,19 @@
+pragma solidity ^0.5.0;
+
+import "../AdminInterface.sol";
+
+
+// A mock implementation of AdminInterface, taking the place of a financial contract.
```

Comment:
I like using a mock financial contract for this instead of the tokenized derivative. I think it makes things much clearer.